x/concentrated-liquidity/position_test.go

```diff
@@ -2245,3 +2246,298 @@ func (s *KeeperTestSuite) TestNegativeTickRange_SpreadFactor() {
 
 	s.clk.InitGenesis(s.Ctx, *export)
 }
+
+func (s *KeeperTestSuite) TestTransferPositions() {
+	// expectedUptimes are used for claimable incentives tests
+	expectedUptimes := getExpectedUptimes()
+
+	errTolerance := osmomath.ErrTolerance{
+		AdditiveTolerance: osmomath.NewDec(1),
+		// Actual should be less than expected, so we round down
+		RoundingDir: osmomath.RoundDown,
+	}
+
+	oldOwner := s.TestAccs[0]
+	newOwner := s.TestAccs[1]
+
+	testcases := map[string]struct {
+		inRangePositions     []uint64
+		outOfRangePositions  []uint64
+		positionsToTransfer  []uint64
+		setupUnownedPosition bool
+		isLastPositionInPool bool
+
+		expectedError error
+	}{
+		"single position ID in range": {
+			inRangePositions:    []uint64{DefaultPositionId},
+			positionsToTransfer: []uint64{DefaultPositionId},
+		},
+		"two position IDs, one in range, one out of range": {
+			inRangePositions:    []uint64{DefaultPositionId},
+			outOfRangePositions: []uint64{DefaultPositionId + 1},
+			positionsToTransfer: []uint64{DefaultPositionId, DefaultPositionId + 1},
+		},
+		"two position IDs in range": {
+			inRangePositions:    []uint64{DefaultPositionId, DefaultPositionId + 1},
+			positionsToTransfer: []uint64{DefaultPositionId, DefaultPositionId + 1},
+		},
+		"three position IDs in range": {
+			inRangePositions:    []uint64{DefaultPositionId, DefaultPositionId + 1, DefaultPositionId + 2},
+			positionsToTransfer: []uint64{DefaultPositionId, DefaultPositionId + 1, DefaultPositionId + 2},
+		},
+		"three position IDs, two in range, one out of range": {
+			inRangePositions:    []uint64{DefaultPositionId, DefaultPositionId + 1},
+			outOfRangePositions: []uint64{DefaultPositionId + 2},
+			positionsToTransfer: []uint64{DefaultPositionId, DefaultPositionId + 1, DefaultPositionId + 2},
+		},
+		"three position IDs, two in range, one out of range, only transfer the two in range": {
+			inRangePositions:    []uint64{DefaultPositionId, DefaultPositionId + 1},
+			outOfRangePositions: []uint64{DefaultPositionId + 2},
```

Comment:
I think we don't need all these cases, most of them seem to be touching the same logic branch of what we want to test thus can be removed. Wdyt? I would personally prefer removing superfluous test cases if not needed

Reply:
I'm generally confused how they differ in output if inputs are the same. What am I missing?

Reply:
@p0mvn what do you mean by differ in outputs if inputs are the same? The inputs should vary across all test cases.

@mattverse I suppose, but the purpose of all these tests is to prove that 
- a user that transfers positions gets the outstanding rewards that the position has prior to transfer
- the positions transfer,
- when those positions accrue rewards it does not go back to them and instead goes to the new owner. 
- the positions that the user owns that they did not transfer remain unaffected

I can get rid of a few of them if you think that is worth the reduced test coverage. I would argue that transfer one in range, transferring one in range and one out of range, and transferring one in range and one out of range while leaving another alone are all ensuring different things

Reply:
I added a comment to the test that explains everything being tested here https://github.com/osmosis-labs/osmosis/pull/6623/commits/b17bd2c9131aef6d05409c563cd2a900bf5644b3

Let me know if that changes your mind WRT removing test cases